### ISSUE TITLE
Support approvals in a template

### ISSUE THREAD

**Issue by jcanizalez** [AzBuilder/terrakube-server#69]
Currently Terraform Enterprise support approvals in a Run

![image](https://user-images.githubusercontent.com/27365102/136992510-2f1e4eb6-ee91-47d9-83fd-738a38fd5003.png)

So add the option to use approvals in Terrakube, once Templates are supported #68 . Add the option to define approvals using a template. In the template an approval will start with `approval:` and the Team name that must approve in order to continue with the execution

Example

YAML
``` yaml
terrakube: plan
approval: AZ_USER
terrakube: apply
approval: AZ_ADMIN
terrakube: destroy
```

API
```
{
  "data": {
    "type": "template",
    "attributes": {
     "id":"template-id",
      "name": "ApplyDestroy",
      "description":"Destroy all the resources after apply successfully"
      "version": "1.0.0",
       "template" : "terrakube: plan
approval: AZ_USER
terrakube: apply
approval: AZ_ADMIN
terrakube: destroy"
    }
  }
}
```

**Comment by ilkerispir** [comment]
Hello @jcanizalez, @alfespa17;

I have a question about this feature. The approval scenario runs successfully. However, in the reject scenario, Terraform apply does not work as expected, but then no plan starts and it waits in pending.

My Plan & Apply Plan with Approval:
```bash
low:
  - type: "terraformPlan"
    name: "Terraform Plan"
    step: 100
  - type: "approval"
    name: "Approval"
    team: "devops@antero.com.au"
    step: 200
  - type: "terraformApply"
    name: "Terraform Apply"
    step: 300
```

Example Screenshots:
![image](https://github.com/AzBuilder/terrakube/assets/37123041/74617130-f856-4c35-8dbc-801641073228)
![image](https://github.com/AzBuilder/terrakube/assets/37123041/402effe2-9fd3-4e6f-9b95-72392319ea68)

**Comment by alfespa17** [comment]
This is a bug n version 2.19.0, I fixed that in 2.19.1, it is related to this

https://github.com/orgs/AzBuilder/discussions/714
